components/rich_text_editor/extensions/emoji/suggestion.js

```diff
@@ -0,0 +1,106 @@
+import { VueRenderer } from '@tiptap/vue-3';
+import { emojisIndexed } from '@/components/emoji_picker/emojis';
+
+import EmojiList from './EmojiList.vue';
+import { createTippy } from '@/components/popover/tippy_utils';
+
+export default {
+  items: ({ query }) => {
+    const emojiList = Object.values(emojisIndexed);
+    const filteredEmoji = emojiList.filter(function (item) {
+      if (this.count < 5 &&
+        item.shortname.substring(1, item.shortname.length - 1).startsWith(query.toLowerCase())) {
+        this.count++;
+        return true;
+      }
+      return false;
+    }, { count: 0 });
+    return filteredEmoji.map(item => item.shortname);
+  },
+
+  command: ({ editor, range, props }) => {
+    // increase range.to by one when the next node is of type "text"
+    // and starts with a space character
+    const nodeAfter = editor.view.state.selection.$to.nodeAfter;
+    const overrideSpace = nodeAfter?.text?.startsWith(' ');
+
+    if (overrideSpace) {
+      range.to += 1;
+    }
+
+    editor
+      .chain()
+      .focus()
+      .insertContentAt(range, [
+        {
+          type: 'emoji',
+          attrs: props,
+        },
+        {
+          type: 'text',
+          text: ' ',
+        },
+      ])
+      .run();
+
+    window.getSelection()?.collapseToEnd();
+  },
+
+  render: () => {
+    let component;
+    let popup;
+
+    return {
+      onStart: props => {
+        component = new VueRenderer(EmojiList, {
+          props,
+          editor: props.editor,
+        });
+
+        if (!props.clientRect) {
+          return;
+        }
+
+        const tippyOptions = {
+          getReferenceClientRect: props.clientRect,
+          appendTo: () => document.body,
```

Comment:
Do we need z-index logic? what if we have a message input inside a popover dialog for example? I think right now the suggestion list would display behind the popover.

Reply:
what value for zIndex would be opt here? I do agree we need some zindex. @iropolo any advice ?

Reply:
Hmm maybe we can use the same value as we have on popper, it is `650`